src/modules/tools/zprobe/ZProbe.cpp

```diff
@@ -240,16 +247,56 @@ bool ZProbe::return_probe(int steps)
     return true;
 }
 
-bool ZProbe::doProbeAt(int &steps, float x, float y)
+bool ZProbe::doProbeAt(int &steps, float x, float y, float actuator_positions[], int repeats)
```

Comment:
adding repeats is a good addition but should be a totally separate pull request.
adding actuator positions is not an interesting change and adds nothing, it is just a different way to do what was already possible.